### PR TITLE
SDK-101 Build goal, and tests

### PR DESCRIPTION
**Pull request by AdamGrzybkowski**

### CODE REVIEW

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/Build.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        if (!tempFolder.exists()) {
+            tempFolder.mkdir();
+        }
+        Model tempModel = createModel();
+        Set<Project> watchedModules = server.getWatchedProjects();
+        for(Project module: watchedModules){
+            Path newLink = Paths.get(new File(tempFolder, module.getArtifactId()).getAbsolutePath());
+            Path existingfile = Paths.get(module.getPath());
+            try {
+                Files.createSymbolicLink(newLink, existingfile);
```
Comment (rkorytkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/Build.java]: It seems createSymbolicLink requires administrator privileges on Windows. We should catch IOException here and simply copy directory if creating symbolic link fails...

File: integration-tests/src/test/resources/integration-test/module1/pom.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  <url>http://maven.apache.org</url>
+  <dependencies>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>3.8.1</version>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+</project>
```
Comment (rkorytkowski) [inline integration-tests/src/test/resources/integration-test/module1/pom.xml]: You must create actual modules using openmrs-sdk:create-platform-module / openmrs-sdk:create-project...

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/Build.java (hunk 1 above)
Comment (AdamGrzybkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/Build.java]: ok, thanks for that :)

Comment (rkorytkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/Build.java]: Maybe setup platform instead of Reference Application to speed it up a bit?

Comment (rkorytkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/Build.java]: Also platform modules should build faster than ref app modules...

Comment (AdamGrzybkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/Build.java]: ok :) I will try it

Comment (AdamGrzybkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/Build.java]: @rkorytkowski building platform modules is much faster :) PR updated

**Pull request merged**